multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/util/ClientKeyConfigurationHandler.java

```diff
@@ -0,0 +1,32 @@
+package org.cloudfoundry.multiapps.controller.persistence.util;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.cloudfoundry.multiapps.controller.persistence.Messages;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClientKeyConfigurationHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientKeyConfigurationHandler.class);
+
+    public void createEncodedKeyFile(String pemKey, String fileName) {
```

Comment:
This method should return the File

Reply:
If this method won't return anything, then change the name of method. `fileName` parameter could be Path object instead String.

Reply:
To clarify If I understand correctly - by returning the File, do you mean returning the Path object returned from the method Files.write()?